src/DotVVM.Framework/Compilation/BindingRequiredResourceVisitor.cs

```diff
@@ -23,61 +23,57 @@ public BindingRequiredResourceVisitor(ControlResolverMetadata requiredResourceCo
-                requiredResources = original;
-            }
+            Visit(view, view.Content, base.VisitView);
         }
 
         public override void VisitControl(ResolvedControl control)
         {
             if (control.Metadata.Type == typeof(Content))
             {
-                var original = requiredResources;
-                base.VisitControl(control);
-                if (original != requiredResources)
-                {
-                    control.Content.AddRange(
-                        requiredResources.Except(original).Select(r => CreateRequiredResourceControl(r, control.DothtmlNode, control.Content.First().DataContextTypeStack)));
-                    requiredResources = original;
-                }
+                Visit(control, control.Content, base.VisitControl);
             }
             else
             {
                 base.VisitControl(control);
             }
         }
 
+        private void Visit<TNodeType>(TNodeType node, List<ResolvedControl> nodeContent, Action<TNodeType> visitBase) where TNodeType : ResolvedTreeNode
+        {
+            var original = requiredResources;
+            visitBase(node);
+            if (original != requiredResources)
+            {
+                nodeContent.AddRange(
+                    requiredResources
+                        .Except(original)
+                        .Select(name => CreateRequiredResourceControl(name, node.DothtmlNode, nodeContent.First().DataContextTypeStack)));
+                requiredResources = original;
+            }
+        }
+
         private ResolvedControl CreateRequiredResourceControl(string resource, Parser.Dothtml.Parser.DothtmlNode node, DataContextStack dataContext)
         {
-            var c = new ResolvedControl(requiredResourceConrolMetadata, node, dataContext);
-            c.SetProperty(new ResolvedPropertyValue(RequiredResource.NameProperty, resource));
-            return c;
+            var control = new ResolvedControl(requiredResourceConrolMetadata, node, dataContext);
+            control.SetProperty(new ResolvedPropertyValue(RequiredResource.NameProperty, resource));
+            return control;
         }
 
         public override void VisitPropertyBinding(ResolvedPropertyBinding propertyBinding)
         {
-            var r = propertyBinding.Binding.Binding.GetProperty<RequiredRuntimeResourcesBindingProperty>(ErrorHandlingMode.ReturnNull);
-            if (r != null)
-                requiredResources = requiredResources.Union(r.Resources);
+            var resourceProperty = propertyBinding.Binding.Binding.GetProperty<RequiredRuntimeResourcesBindingProperty>(ErrorHandlingMode.ReturnNull);
+            if (resourceProperty != null)
+            {
+                requiredResources = requiredResources.Union(resourceProperty.Resources);
```

Comment:
I'm sorry, but this change does not make it more readable, I actually struggle to visually recognize. I'd argue that single-letter name like `r` is completely reasonable in this context.

```
resourceProperty and requiredResources
```

Reply:
I admit that in this situation, more descriptive naming of variable doesn't add too much, but in the future you can add more statements to the function and from single-letter name like `r` you will not be able to determine what the variable represents. If you struggle with recognizing `resourceProperty` and `requiredResources` I can change it to `property` if you find to more recognizable.

Reply:
That would be better for visual recognision but even less descriptive than `r`. In this context it's totally unimportant that it's a binding property, the important fact is that it contains the  resource requirements of this binding.

Reply:
ok, so in this situation I'd choose `requiredResourceProperty`. @djanosik what do you think?